pkg/replicas/getter.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// the code below implements the integration of dispersed replicas in chunk fetching.
+// using storage.Getter interface.
+package replicas
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/file/redundancy"
+	"github.com/ethersphere/bee/pkg/soc"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// ErrSwarmageddon is returned in case of a vis mayor called Swarmageddon.
+// Swarmageddon is the situation when none of the replicas can be retrieved.
+// If 2^{depth} replicas were uploaded and they all have valid postage stamps
+// then the probability of Swarmageddon is less than 0.000001
+// assuming the error rate of chunk retrievals stays below the level expressed
+// as depth by the publisher.
+type ErrSwarmageddon struct {
```

Comment:
This error type doesn't add much clarity since the `errors.Join` can be returned directly and checked with `errors.Is` without wrapping it to an extra layer. Also, the error is only used in the tests.

Reply:
Still unresolved.